### PR TITLE
dev/reference/tools, TOC: first batch of renaming reference files

### PR DESCRIPTION
**Pull request by dcalvin**
First batch of file renaming and re-positioning of Reference part, including:

- TOC
- Tools

### CODE REVIEW

File: dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md [hunk 1]
```diff
@@ -243,7 +244,7 @@ MySQL 及 DM 操作与处理流程如下：
         » unlock-ddl-lock test-`shard_db`.`shard_table`
         {
             "result": false,
-            "msg": "github.com/pingcap/tidb-enterprise-tools/dm/master/server.go:1472: DDL lock test-`shard_db`.`shard_table` owner ExecuteDDL successfully, so DDL lock removed. but some dm-workers ExecuteDDL fail, you should to handle dm-worker directly",
+            "msg": "github.com/pingcap/tidb-enterprise-dev/reference/tools/data-migration/master/server.go:1472: DDL lock test-`shard_db`.`shard_table` owner ExecuteDDL successfully, so DDL lock removed. but some dm-workers ExecuteDDL fail, you should to handle dm-worker directly",
```
Comment (dcalvin) [inline dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md]: "msg": "github.com/pingcap/tidb-enterprise-tools/dm/master/server.go:1472: DDL lock test-`shard_db`.`shard_table` owner ExecuteDDL successfully, so DDL lock removed. but some dm-workers ExecuteDDL fail, you should to handle dm-worker directly",

File: dev/reference/tools/data-migration/precheck.md [hunk 2]
```diff
@@ -40,7 +41,7 @@ category: tools
     TiDB 和 MySQL 的兼容性存在以下一些区别：
 
     - TiDB 不支持外键
-    - 字符集的兼容性不同，详见 [TiDB 支持的字符集](/sql/character-set-support.md)
+    - 字符集的兼容性不同，详见 [TiDB 支持的字符集](dev/reference/sql/character-set.md)
```
Comment (dcalvin) [inline dev/reference/tools/data-migration/precheck.md]:
```suggestion
    - 字符集的兼容性不同，详见 [TiDB 支持的字符集](/dev/reference/sql/character-set.md)
```

File: dev/reference/tools/mydumper.md [hunk 3]
```diff
@@ -10,12 +11,12 @@ category: tools
 
 `mydumper` 是 [mydumper](https://github.com/maxbube/mydumper) 的 fork 项目，并添加了一些针对 TiDB 的功能。推荐使用此工具对 TiDB 进行逻辑备份。
 
-mydumper 包含在 tidb-enterprise-tools 安装包中，可[在此下载](../tools/download.md)。
+mydumper 包含在 tidb-enterprise-tools 安装包中，可[在此下载](/dev/reference/tool/download.md)。
```
Comment (dcalvin) [inline dev/reference/tools/mydumper.md]:
```suggestion
mydumper 包含在 tidb-enterprise-tools 安装包中，可[在此下载](/dev/reference/tools/download.md)。
```

File: dev/reference/tools/tidb-binlog/monitor.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ---
 title: TiDB-Binlog 集群监控
-category: tools
+category: reference
+aliases: ['/docs-cn/tools/binlog/monitor']
```
Comment (dcalvin) [inline dev/reference/tools/tidb-binlog/monitor.md]:
```suggestion
aliases: ['/docs-cn/tools/binlog/monitor/']
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ---
 title: TiDB-Binlog 集群运维
-category: tools
+category: reference
+aliases: ['/docs-cn/tools/binlog/operation']
```
Comment (dcalvin) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
aliases: ['/docs-cn/tools/binlog/operation/']
```

File: dev/reference/tools/tidb-binlog/overview.md [hunk 6]
```diff
@@ -1,7 +1,7 @@
 ---
 title: TiDB-Binlog 简介
-category: tools
-aliases: ['/docs-cn/tools/tidb-binlog-cluster/']
+category: reference
+aliases: ['/docs-cn/tools/binlog/overview']
```
Comment (dcalvin) [inline dev/reference/tools/tidb-binlog/overview.md]:
```suggestion
aliases: ['/docs-cn/tools/binlog/overview/']
```

File: dev/reference/tools/tidb-binlog/upgrade.md [hunk 7]
```diff
@@ -1,11 +1,12 @@
 ---
 title: TiDB-Binlog Cluster 版本升级方法
-category: tools
+category: reference
+aliases: ['/docs-cn/tools/binlog/upgrade']
```
Comment (dcalvin) [inline dev/reference/tools/tidb-binlog/upgrade.md]:
```suggestion
aliases: ['/docs-cn/tools/binlog/upgrade/']
```

File: dev/reference/tools/tidb-binlog/upgrade.md [hunk 8]
```diff
@@ -1,11 +1,12 @@
...
 ---
 
 # TiDB-Binlog Cluster 版本升级方法
 
-新版本的 TiDB（v2.0.8-binlog、v2.1.0-rc.5 及以上版本）不兼容 [Kafka 版本](/tools/binlog/tidb-binlog-kafka.md)以及 [Local 版本](/tools/binlog/tidb-binlog-local.md)的 TiDB-Binlog，集群升级到新版本后只能使用 Cluster 版本的 TiDB-Binlog。如果在升级前已经使用了 Kafka／Local 版本的 TiDB-Binlog，必须将其升级到 Cluster 版本。
+新版本的 TiDB（v2.0.8-binlog、v2.1.0-rc.5 及以上版本）不兼容 [Kafka 版本](/dev/reference/tidb-binlog/tidb-binlog-kafka.md)以及 [Local 版本](/dev/reference/tidb-binlog/tidb-binlog-local.md)的 TiDB-Binlog，集群升级到新版本后只能使用 Cluster 版本的 TiDB-Binlog。如果在升级前已经使用了 Kafka／Local 版本的 TiDB-Binlog，必须将其升级到 Cluster 版本。
```
Comment (dcalvin) [inline dev/reference/tools/tidb-binlog/upgrade.md]:
```suggestion
新版本的 TiDB（v2.0.8-binlog、v2.1.0-rc.5 及以上版本）不兼容 [Kafka 版本](/dev/reference/tools/tidb-binlog/tidb-binlog-kafka.md)以及 [Local 版本](/dev/reference/tools/tidb-binlog/tidb-binlog-local.md)的 TiDB-Binlog，集群升级到新版本后只能使用 Cluster 版本的 TiDB-Binlog。如果在升级前已经使用了 Kafka／Local 版本的 TiDB-Binlog，必须将其升级到 Cluster 版本。
```

File: dev/reference/tools/tidb-binlog/upgrade.md [hunk 9]
```diff
@@ -19,7 +20,7 @@ TiDB-Binlog 版本与 TiDB 版本的对应关系如下：
 
 > **注意：**
 >
-> 如果能接受重新导全量数据，则可以直接废弃老版本，按 [TiDB-Binlog 集群部署](/tools/binlog/deploy.md)中的步骤重新部署。
+> 如果能接受重新导全量数据，则可以直接废弃老版本，按 [TiDB-Binlog 集群部署](/dev/reference/tidb-binlog/deploy.md)中的步骤重新部署。
```
Comment (dcalvin) [inline dev/reference/tools/tidb-binlog/upgrade.md]:
```suggestion
> 如果能接受重新导全量数据，则可以直接废弃老版本，按 [TiDB-Binlog 集群部署](/dev/reference/tools/tidb-binlog/deploy.md)中的步骤重新部署。
```

File: dev/reference/tools/data-migration/cluster-operations.md [hunk 10]
```diff
@@ -1,6 +1,7 @@
 ---
 title: DM 集群操作
-category: tools
+category: reference
+aliases: ['/docs-cn/tools/dm/cluster-operations']
```
Comment (dcalvin) [inline dev/reference/tools/data-migration/cluster-operations.md]:
```suggestion
aliases: ['/docs-cn/tools/dm/cluster-operations/']
```

File: dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md (hunk 1 above)
Comment (dcalvin) [inline dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md]:
```suggestion
      "msg": "github.com/pingcap/tidb-enterprise-tools/dm/master/server.go:1472: DDL lock test-`shard_db`.`shard_table` owner ExecuteDDL successfully, so DDL lock removed. but some dm-workers ExecuteDDL fail, you should to handle dm-worker directly",
```

File: dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md [hunk 11]
```diff
@@ -243,7 +244,7 @@ MySQL 及 DM 操作与处理流程如下：
         » unlock-ddl-lock test-`shard_db`.`shard_table`
         {
             "result": false,
-            "msg": "github.com/pingcap/tidb-enterprise-tools/dm/master/server.go:1472: DDL lock test-`shard_db`.`shard_table` owner ExecuteDDL successfully, so DDL lock removed. but some dm-workers ExecuteDDL fail, you should to handle dm-worker directly",
+      "msg": "github.com/pingcap/tidb-enterprise-tools/dm/master/server.go:1472: DDL lock test-`shard_db`.`shard_table` owner ExecuteDDL successfully, so DDL lock removed. but some dm-workers ExecuteDDL fail, you should to handle dm-worker directly",
```
Comment (dcalvin) [inline dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md]:
```suggestion
            "msg": "github.com/pingcap/tidb-enterprise-tools/dm/master/server.go:1472: DDL lock test-`shard_db`.`shard_table` owner ExecuteDDL successfully, so DDL lock removed. but some dm-workers ExecuteDDL fail, you should to handle dm-worker directly",
```

Comment (dcalvin) [inline dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md]:
```suggestion
           "msg": "github.com/pingcap/tidb-enterprise-tools/dm/master/server.go:1472: DDL lock test-`shard_db`.`shard_table` owner ExecuteDDL successfully, so DDL lock removed. but some dm-workers ExecuteDDL fail, you should to handle dm-worker directly",
```

File: dev/reference/tools/tidb-binlog/deploy.md [hunk 12]
```diff
@@ -1,6 +1,7 @@
 ---
 title: TiDB-Binlog 集群部署
-category: tools
+category: reference
+aliases: ['/docs-cn/tools/binlog/deploy']
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/deploy.md]:
```suggestion
aliases: ['/docs-cn/tools/binlog/deploy/']
```

File: dev/reference/tools/data-migration/precheck.md [hunk 13]
```diff
@@ -40,7 +41,7 @@ category: tools
     TiDB 和 MySQL 的兼容性存在以下一些区别：
 
     - TiDB 不支持外键
-    - 字符集的兼容性不同，详见 [TiDB 支持的字符集](/sql/character-set-support.md)
+    - 字符集的兼容性不同，详见 [TiDB 支持的字符集](/dev/reference/sql/character-set.md)
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply dev/reference/tools/data-migration/precheck.md]: In next PR...

File: dev/reference/tools/data-migration/skip-replace-sqls.md [hunk 14]
```diff
@@ -1,13 +1,14 @@
...
+category: reference
+aliases: ['/docs-cn/tools/dm/skip-replace-sqls/']
 ---
 
 # 跳过 (skip) 或替代执行 (replace) 异常的 SQL 语句
 
 本文介绍了如何使用 DM 来处理异常的 SQL 语句。
 
-目前，TiDB 并不完全兼容所有的 MySQL 语法（详见 [TiDB 已支持的 DDL 语句](/sql/ddl.md)）。当使用 DM 从 MySQL 同步数据到 TiDB 时，如果 TiDB 不支持对应的 SQL 语句，可能会造成错误并中断同步任务。在这种情况下，DM 提供以下两种方式来恢复同步：
+目前，TiDB 并不完全兼容所有的 MySQL 语法（详见 [TiDB 已支持的 DDL 语句](/dev/reference/sql/ddl.md)）。当使用 DM 从 MySQL 同步数据到 TiDB 时，如果 TiDB 不支持对应的 SQL 语句，可能会造成错误并中断同步任务。在这种情况下，DM 提供以下两种方式来恢复同步：
```
Comment (dcalvin) [inline dev/reference/tools/data-migration/skip-replace-sqls.md]:
```suggestion
目前，TiDB 并不完全兼容所有的 MySQL 语法（详见 [TiDB 已支持的 DDL 语句](/dev/reference/sql/statements/ddl.md)）。当使用 DM 从 MySQL 同步数据到 TiDB 时，如果 TiDB 不支持对应的 SQL 语句，可能会造成错误并中断同步任务。在这种情况下，DM 提供以下两种方式来恢复同步：
```

[the remark this answers is not in the document]
Comment (dcalvin) [reply dev/reference/tools/data-migration/skip-replace-sqls.md]: In next PR.

File: dev/reference/tools/data-migration/overview.md [hunk 15]
```diff
@@ -82,20 +83,20 @@ DM 支持对原分库分表进行合库合表操作，但需要满足一些[使
     > - 5.7.1 < MySQL 版本 < 5.8
     > - MariaDB 版本 >= 10.1.3
 
-    在使用 dmctl 启动任务时，DM 会自动对任务上下游数据库的配置、权限等进行[前置检查](/tools/dm/precheck.md)。
+    在使用 dmctl 启动任务时，DM 会自动对任务上下游数据库的配置、权限等进行[前置检查](/dev/reference/tools/data-migration/precheck.md)。
 
 + DDL 语法
 
-    - 目前，TiDB 部分兼容 MySQL 支持的 DDL 语句。因为 DM 使用 TiDB parser 来解析处理 DDL 语句，所以目前仅支持 TiDB parser 支持的 DDL 语法。详见 [TiDB DDL 语法支持](../../sql/ddl.md)。
+    - 目前，TiDB 部分兼容 MySQL 支持的 DDL 语句。因为 DM 使用 TiDB parser 来解析处理 DDL 语句，所以目前仅支持 TiDB parser 支持的 DDL 语法。详见 [TiDB DDL 语法支持](/dev/reference/sql/ddl.md)。
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply dev/reference/tools/data-migration/overview.md]: In next ...

File: dev/reference/tools/data-migration/precheck.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (lilin90) [reply dev/reference/tools/data-migration/precheck.md]: @dcalvin Got it.

File: dev/reference/tools/data-migration/overview.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (lilin90) [reply dev/reference/tools/data-migration/overview.md]: OK.

**Comment by dcalvin** [comment]
Resolved further conflicts brought by https://github.com/pingcap/docs-cn/pull/1334/files. And updated links where necessary. Think this PR is ready to go.  PTAL @lilin90

**Comment by lilin90** [comment]
@dcalvin Great! I have finished my review on #1351. If you want to merge this PR after #1351 is ready, you can address those comments asap.

File: dev/reference/tools/tidb-lightning/deployment.md [hunk 16]
```diff
@@ -48,11 +49,11 @@ category: tools
 >
 > - `tikv-importer` 将中间数据存储缓存到内存上以加速导入过程。占用内存大小可以通过 **(`max-open-engines` × `write-buffer-size` × 2) + (`num-import-jobs` × `region-split-size` × 2)** 计算得来。如果磁盘写入速度慢，缓存可能会带来更大的内存占用。
 
-此外，目标 TiKV 集群必须有足够空间接收新导入的数据。除了[标准硬件配置](../../op-guide/recommendation.md)以外，目标 TiKV 集群的总存储空间必须大于 **数据源大小 × [副本数量](../../FAQ.md#3-2-6-%E6%AF%8F%E4%B8%AA-region-%E7%9A%84-replica-%E6%95%B0%E9%87%8F%E5%8F%AF%E9%85%8D%E7%BD%AE%E5%90%97-%E8%B0%83%E6%95%B4%E7%9A%84%E6%96%B9%E6%B3%95%E6%98%AF) × 2**。例如集群默认使用 3 副本，那么总存储空间需为数据源大小的 6 倍以上。
+此外，目标 TiKV 集群必须有足够空间接收新导入的数据。除了[标准硬件配置](/dev/how-to/deploy/hardware-recommendations.md)以外，目标 TiKV 集群的总存储空间必须大于 **数据源大小 × [副本数量](/FAQ.md#3-2-6-%E6%AF%8F%E4%B8%AA-region-%E7%9A%84-replica-%E6%95%B0%E9%87%8F%E5%8F%AF%E9%85%8D%E7%BD%AE%E5%90%97-%E8%B0%83%E6%95%B4%E7%9A%84%E6%96%B9%E6%B3%95%E6%98%AF) × 2**。例如集群默认使用 3 副本，那么总存储空间需为数据源大小的 6 倍以上。
```
Comment (lilin90) [inline dev/reference/tools/tidb-lightning/deployment.md]:
```suggestion
此外，目标 TiKV 集群必须有足够空间接收新导入的数据。除了[标准硬件配置](/dev/how-to/deploy/hardware-recommendations.md)以外，目标 TiKV 集群的总存储空间必须大于 **数据源大小 × [副本数量](/FAQ.md#3-2-6-每个-region-的-replica-数量可配置吗-调整的方法是) × 2**。例如集群默认使用 3 副本，那么总存储空间需为数据源大小的 6 倍以上。
```

**Pull request merged**